sycl/source/detail/platform_util.cpp

```diff
@@ -41,37 +43,6 @@ uint32_t PlatformUtil::getMaxClockFrequency() {
   throw runtime_error(
       "max_clock_frequency parameter is not supported for host device",
       PI_ERROR_INVALID_DEVICE);
-#if defined(__x86_64__) || defined(__i386__)
```

Comment:
Would it be useful to enable that dead code rather than removing it?

Reply:
According to the exception message above, it seems like this function is only used for host device, which we recently made inaccessible for users. Therefore, I don't see a value in enabling this code and it could be the case that we can perform even more cleanup (but I would prefer to do that in a separate PR and it could be ABI-breaking)